### PR TITLE
removed email from db

### PR DESCRIPTION
**Pull request by interestIngc**
## PR overview

I have removed email from the registration form as well as from the database.
Instead of requiring email at the registration stage, we will now require it when the user initiates the verification process (in the current flow this happens when user clicks the "Verify Email" button).

To implement this, I have added a new html page "input-your-email.html" which contains a form field to enter the email address and a "Get Code" button. It shows up after a user clicks the "Verify Email" button.
After clicking the "Get Code" button, the user is directed to the "input-verification-code.html" page, where they are asked to enter the verification code sent to the provided email address.

A new endpoint "/input-your-email-page" was implemented, which returns "input-your-email.html".
Also I renamed "/verify-email-page" -> "/input-verification-code-page".

Removed email field from "profile.html" and "register.html".

Removed email column from the "users" table.

### SECTION 1: RESOURCE SERVER (http://localhost:5001/)
- [x] http://localhost:5001/ home page loads
- [x] "LoginAsUser" button takes you to the user login page
- [x] Log in using "tester" and "12341234" should succeed
- [x] Updating the display name should work
- [x] Log out and Register a new user and try to log in with that new user
- [x] "LoginAsClient" button takes you to the client login page
- [x] Log in using "layer8" and "12341234" should work
- [x] You will be able to see your UID and Secret in profile
- [x] Log out and Register a new client and log in using that should work

### SECTION 2: WE'VE GOT POEMS (http://localhost:5173/)
- [x] Log in with 'tester' / '1234'
- [x] Log in anonymously with Layer8
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login screen
- [x] Clicking "Register" takes you to the registration page
- [x] Registering with a username, password, and profile image is successful
- [x] Logging in with the new username / password succeeds
- [x] Logging in with Layer8 opens the pop-up
- [x] Logging in with the newly registered credentials from Section 1 succeeds
- [x] User chooses to share their new "Username" & "Country" from the Layer8 Resource Server
- [x] Clicking "Get Next Poem" loads different poem correctly x 3
- [x] Clicking "Logout" takes you to the login page
- [x] Log in with 'tester' / '1234', and then with the credentials from Section 1 succeeds
- [x] This time, DO NOT share "display name" or "country"
- [x] Clicking "Get Next Poem" loads different poem correctly x 3

### SECTION 3: IMSHARER (http://localhost:5174/)
- [x] Main page loads
- [x] Upload of image works
- [x] Reload leads to instant reload (demonstrating proper caching)
- [x] Clicking the newly loaded image shows it in a lightbox

### CODE REVIEW

File: server/resource_server/repository/repository_test.go [hunk 1]
```diff
@@ -160,7 +159,17 @@ func TestLoginPreCheckUser(t *testing.T) {
 	}
 
 	// Expect a query to be executed and return a row
-	mock.ExpectQuery(regexp.QuoteMeta(`SELECT * FROM "users" WHERE username = $1 ORDER BY "users"."id" LIMIT 1`)).WithArgs(testLoginPrecheck.Username).WillReturnRows(sqlmock.NewRows([]string{"id", "email", "username", "first_name", "last_name", "password", "salt"}).AddRow(1, "user@test.com", "test_user", "Test", "User", "testpass", "testsalt"))
+	mock.ExpectQuery(
```
Comment (stravid87) [inline server/resource_server/repository/repository_test.go:162]: Eventually you will be the owner of this function. May as well do these edits now 👍

File: server/resource_server/service/service.go [hunk 2]
```diff
@@ -177,15 +172,15 @@ func (s *service) ProfileClient(userName string) (models.ClientResponseOutput, e
 	return clientModel, nil
 }
 
-func (s *service) VerifyEmail(userID uint) error {
+func (s *service) VerifyEmail(userID uint, userEmail string) error {
```
Comment (stravid87) [inline server/resource_server/service/service.go:175]: Hmmmmm, there should definitely be a Mandelbrot Set emoji that I can insert here...

**Review by stravid87 — APPROVED** [review]
Great work. We are one step closer!

**Review by huzaifamk — APPROVED** [review]
All looks good, just a reply to a comment added, approved.

File: server/resource_server/controller/controller.go [hunk 3]
```diff
@@ -225,11 +229,30 @@ func VerifyEmailHandler(w http.ResponseWriter, r *http.Request) {
 	tokenString = tokenString[7:] // Remove the "Bearer " prefix
 	userID, err := utils.ValidateToken(tokenString)
 	if err != nil {
-		utils.HandleError(w, http.StatusBadRequest, "Failed to verify email", err)
+		utils.HandleError(w, http.StatusBadRequest, "Request failed: invalid authorization token", err)
```
[the remark this answers is not in the document]
Comment (interestIngc) [reply server/resource_server/controller/controller.go:232]: Thanks for the reference! I agree that token validation should be extracted into a separate handler or utility function. Currently we just copy-paste this between request handler functions. I think that should be done in a separate PR, as it is related not only to this, but to all the handlers. I'll create an issue with that)

File: server/assets-v1/templates/src/pages/user_portal/email/input-verification-code.html [hunk 4]
```diff
@@ -67,12 +73,15 @@
...
                             code: verificationCode.value,
                         }),
-                    }
+                    },
                 );
                 const result = await response.json();
 
+                localStorage.removeItem("email");
+
                 alert(result.message);
```
[the remark this answers is not in the document]
Comment (interestIngc) [reply server/assets-v1/templates/src/pages/user_portal/email/input-verification-code.html:85]: Did that)
Screenshot 2024-06-16 at 23 29 15

**Comment by ternakkode** [comment]
We're not supposed to change the existing migration but instead create a new migration to drop the email from users table

**Pull request merged**